lms/services/digest.py

```diff
@@ -165,6 +210,48 @@ def instructor_digest(self, h_userid):
             "courses": course_digests,
         }
 
+    @property
+    def assignment_infos(self):
+        """Return the list of AssignmentInfo's for all the assignment IDs in self.annotations."""
+        if self._assignment_infos is None:
+            self._assignment_infos = [
+                AssignmentInfo(
+                    row.id,
+                    row.tool_consumer_instance_guid,
+                    row.resource_link_id,
+                    row.title,
+                    row.authority_provided_id,
+                )
+                for row in self._db.execute(
+                    select(
+                        Assignment.id,
+                        Assignment.tool_consumer_instance_guid,
+                        Assignment.resource_link_id,
+                        Assignment.title,
+                        Course.authority_provided_id,
+                    ).where(
```

Comment:
There are assignments without titles, I reckon we should filter those out here and let the fallback to courses take care of that.

Reply:
Fixed

Reply:
Looks like we have both version of no title, null and `''`

https://report.hypothes.is/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7InR5cGUiOiJuYXRpdmUiLCJuYXRpdmUiOnsicXVlcnkiOiJzZWxlY3QgY3JlYXRlZCwgdGl0bGUsIHRpdGxlIGlzIG51bGwgIGFzIG5vbmUsIGJvb2wodGl0bGUgPSAnJykgZW1wdHkgZnJvbSBhc3NpZ25tZW50IHdoZXJlIHRpdGxlIGlzIG51bGwgb3IgdGl0bGUgPSAnJ1xub3JkZXIgYnkgY3JlYXRlZCBkZXNjIiwidGVtcGxhdGUtdGFncyI6e319LCJkYXRhYmFzZSI6NX0sImRpc3BsYXkiOiJ0YWJsZSIsInBhcmFtZXRlcnMiOltdLCJ2aXN1YWxpemF0aW9uX3NldHRpbmdzIjp7fX0=

We don't do any normalization of what we  get from the LMS here, not sure if we should tringto coerce all those '' to None.

Anyway, for now we'd need a `or title <> ''` here too.